apps/rush-lib/src/logic/InstallManager.ts

```diff
@@ -176,19 +197,22 @@ export class InstallManager {
     return this._commonNodeModulesMarker;
   }
 
-  constructor(rushConfiguration: RushConfiguration, purgeManager: PurgeManager) {
+  constructor(rushConfiguration: RushConfiguration, purgeManager: PurgeManager, options: IInstallManagerOptions) {
     this._rushConfiguration = rushConfiguration;
     this._commonTempFolderRecycler = purgeManager.commonTempFolderRecycler;
+    this._options = options;
 
     this._commonNodeModulesMarker = new LastInstallFlag(this._rushConfiguration.commonTempFolder, {
       node: process.versions.node,
       packageManager: rushConfiguration.packageManager,
-      packageManagerVersion: rushConfiguration.packageManagerToolVersion
+      packageManagerVersion: rushConfiguration.packageManagerToolVersion,
+      variant: options.variant || ''
```

Comment:
>|| '' [](start = 31, length = 5)

is it not OK to just use undefined here?

Reply:
This object gets JSON-serialized, and `undefined` is not a valid value. Also, `null` is forbidden by TSLint, so I elected to use the empty string since it still works for 'default variant'.